tests/components/modbus/conftest.py

```diff
@@ -113,7 +101,24 @@ async def mock_modbus(
 
 
 @pytest.fixture
-async def mock_do_cycle(hass):
+async def mock_pymodbus_return(hass, register_words, mock_modbus):
+    """Trigger update call with time_changed event."""
+    if register_words is None:
```

Comment:
My suggestion was that instead of setting the side effect implicitly depending on `register_words` we have a separate fixture that returns the side effect. Then we use that side effect here, always. The default side effect can be `None`.

Reply:
The side effect of your suggestion is that some of the tests need to be divided in 2 (one to test return values, and one to test exceptions), but it might end up being less complicated, let me try that.

Reply:
Please have a look at e.g. test_binary_sensor.py line 106, that is how I use it at the moment. I do not really understand how you would parametrize your idea (I need something to tell when not to use the default).

Reply:
exception is its own fixture, with a default parameter, and added pytest.mark.parametrize where needed.